Assignments/Program.cs

```diff
@@ -12,35 +13,34 @@ public class T15 {
    public static void Main () {
       Console.WriteLine ("\x1B[4m" + "Check for Isogram Word:-" + "\x1B[0m");
       Console.WriteLine ("An isogram is a word that has no duplicate letters.");
-      string output;
-      List<string> testCases = new () { "apple", "banana", "cat", "doctrine", "elephant", "fruits", "pebble", "rubiks", "yellow", "zebra" };
+      List<string> testCases = new () { "", "apple", "banana", "cat", "doctrine", "elephant", "fruits", "pebble", "rubiks", "yellow", "zebra", "12345", "123345", "@#$4%! -)", "*765^%$*" };
       Console.WriteLine ();
       Console.WriteLine ("\x1B[4m" + "Test Cases:" + "\x1B[0m");
       for (int i = 0; i < testCases.Count; i++) {
-         Console.Write ($"{i + 1}. {testCases[i]} - ");
-         (output, Console.ForegroundColor) = IsIsogram (testCases[i]) ? ("Isogram", Green) : ("Not an Isogram", Red);
-         Console.WriteLine ($"{output}");
-         Console.ResetColor ();
+         var t = testCases[i];
+         Console.Write ($"{i + 1, 2}. {t, 11} - ");
+         perform (t);
       }
       Console.Write ("\nNow, let's enter a word: ");
-      string input = Console.ReadLine ().ToUpper ();
-      (output, Console.ForegroundColor) = IsIsogram (input) ? ("Isogram", Green) : ("Not an Isogram", Red);
+      perform (Console.ReadLine ());
+   }
+
+   /// <summary>Prints the result whether the input is isogram or not</summary>
+   public static void perform (string input) {
+      (string output, Console.ForegroundColor) = IsIsogram (input) ? ("Isogram", Green) : ("Not an Isogram", Red);
       Console.WriteLine ($"{output}");
       Console.ResetColor ();
    }
 
    /// <summary>Returns true if input string is an isogram, otherwise false</summary>
    public static bool IsIsogram (string input) {
+      if (input.Length == 0) return false;
       input = input.ToUpper ();
-      bool isIsogram = true;
-      if (input.All (char.IsLetter)) {
-         List<int> HashTable = new ();
-         for (int j = 0; j < 26; j++) HashTable.Add (0);
-         foreach (char c in input) {
-            if (HashTable[c - 65] == 0) HashTable[c - 65]++;
-            else isIsogram = false;
-         }
+      Dictionary<char, int> freq = new ();
+      foreach (char c in input) {
+         if (freq.TryGetValue (c, out int _)) return false;
+         else freq.Add (c, 1);

```

Comment:
As the if condition fails this line will automatically be executed so else not required.

Reply:
If the ELSE condition is removed, there will be no key-value pairs added to the dict.  So the if condition always fails, and this IsIsogram method always returns TRUE except for the empty string.
![image](https://github.com/vikrampg1710-tmm/Training/assets/117505871/c3e6c063-4f05-4c3b-9730-a7c43e66fc01)

Reply:
@vikrampg1710-tmm  You misunderstood. I did mean just the _else_. Not the entire line.